### PR TITLE
Increase test coverage for SAMReadGroupRecord

### PR DESCRIPTION
**Pull request by magicDGS**
### Description

Explicit and/or new tests for SAMReadGroupRecord

### Checklist

- [x] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: src/test/java/htsjdk/samtools/SAMReadGroupRecordTest.java [hunk 1]
```diff
@@ -40,4 +46,74 @@ public void testGetSAMString() {
...
+
+    @Test
+    public void testGettersAndSetters() throws Exception {
+        final SAMReadGroupRecord rg = new SAMReadGroupRecord("rg1");
+        // test the ID from the two getters
+        Assert.assertEquals(rg.getId(), "rg1");
+        Assert.assertEquals(rg.getReadGroupId(), "rg1");
+
+        // test that the rest of getters returns null if unset
+        testGetterAndSetter(rg, "sample", SAMReadGroupRecord::setSample, SAMReadGroupRecord::getSample);
```
Comment (lbergelson) [inline src/test/java/htsjdk/samtools/SAMReadGroupRecordTest.java]: I think these cases should probably be extracted into a dataprovider.

Comment (magicDGS) [inline src/test/java/htsjdk/samtools/SAMReadGroupRecordTest.java]: Done.

**Comment by magicDGS** [comment]
Changed to a data provider, @lbergelson - back to you and thanks!

**Comment by lbergelson** [comment]
@magicDGS Thank you!

**Pull request merged**